bindings/java/src/org/sleuthkit/datamodel/TaggingManager.java

```diff
@@ -0,0 +1,458 @@
+ */
+package org.sleuthkit.datamodel;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import org.sleuthkit.datamodel.SleuthkitCase.CaseDbConnection;
+
+/**
+ * Provides an API to manage Tags.
+ */
+public class TaggingManager {
+	private final SleuthkitCase skCase;
+
+	/**
+	 * Construct a TaggingManager for the given SleuthkitCase.
+	 *
+	 * @param skCase The SleuthkitCase.
+	 */
+	TaggingManager(SleuthkitCase skCase) {
+		this.skCase = skCase;
+	}
+
+	/**
+	 * Returns a list of all the TagSet that exist in the case.
+	 *
+	 * @return A List of TagSet objects or an empty list if none were found.
+	 *
+	 * @throws TskCoreException
+	 */
+	public List<TagSet> getTagSets() throws TskCoreException {
+		List<TagSet> tagSetList = new ArrayList<>();
+		CaseDbConnection connection = skCase.getConnection();
+		skCase.acquireSingleUserCaseWriteLock();
+		String getAllTagSetsQuery = "SELECT * FROM tag_sets";
+		try (Statement stmt = connection.createStatement(); ResultSet resultSet = stmt.executeQuery(getAllTagSetsQuery);) {
+			while (resultSet.next()) {
+				int setID = resultSet.getInt("tag_set_id");
+				String setName = resultSet.getString("name");
+				TagSet set = new TagSet(setID, setName);
+				set.addTagNames(getTagNamesByTagSetID(setID));
+				tagSetList.add(set);
+			}
+		} catch (SQLException ex) {
+			throw new TskCoreException("Error occurred getting TagSet list.", ex);
+		} finally {
+			connection.close();
+			skCase.releaseSingleUserCaseWriteLock();
+		}
+		return tagSetList;
+	}
+
+	/**
+	 * Inserts a row into the reports table in the case database.
+	 *
+	 * @param name The tag set name.
+	 *
+	 * @return A TagSet object for the new row.
+	 *
+	 * @throws TskCoreException
+	 */
+	public TagSet addTagSet(String name) throws TskCoreException {
+		if (name == null || name.isEmpty()) {
+			throw new IllegalArgumentException("Error adding TagSet, TagSet name must be non-empty string.");
+		}
+
+		TagSet tagSet = null;
+
+		CaseDbConnection connection = skCase.getConnection();
+		skCase.acquireSingleUserCaseWriteLock();
+		try (Statement stmt = connection.createStatement()) {
+			connection.beginTransaction();
+			// INSERT INTO tag_sets (name) VALUES('%s')
+			stmt.execute(String.format("INSERT INTO tag_sets (name) VALUES('%s')", name));
+
+			try (ResultSet resultSet = stmt.getGeneratedKeys()) {
+
+				resultSet.next();
+				int setID = resultSet.getInt(1);
+
+				tagSet = new TagSet(setID, name);
+			}
+			connection.commitTransaction();
+		} catch (SQLException ex) {
+			connection.rollbackTransaction();
+			throw new TskCoreException(String.format("Error adding report %s", name), ex);
+		} finally {
+			connection.close();
+			skCase.releaseSingleUserCaseWriteLock();
+		}
+
+		return tagSet;
+	}
+
+	/**
+	 * Remove a row from the tag set table. All tags belonging to the given tag
+	 * set will also be removed.
```

Comment:
We should actually call out that the tag names in the set will not be deleted, nor will any tags with the tag names from the deleted tag set be deleted.